flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/source/FileSourceFunction.java

```diff
@@ -46,39 +51,25 @@ public FileSourceFunction(InputFormat<String, ?> format, TypeInformation<String>
 	}
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public void open(Configuration parameters) throws Exception {
 		StreamingRuntimeContext context = (StreamingRuntimeContext) getRuntimeContext();
 		this.provider = context.getInputSplitProvider();
 		inputFormat.configure(context.getTaskStubParameters());
+		serializer = typeInfo.createSerializer(getRuntimeContext().getExecutionConfig());
```

Comment:
As far as I understand the streaming code, the execution config at runtime is always invalid.

When running a streaming job, you'll get plenty of error messages like this:
```
11:26:25,600 WARN  org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable     - Environment did not contain an ExecutionConfig - using a default config.
```
I was told that Streaming doesn't need the EC at runtime because all serializers are created at the client and then send over serialized.